lib/recording/recording_jobs.ex

```diff
@@ -0,0 +1,150 @@
+defmodule Onvif.Recording.RecordingJob do
+  @moduledoc """
+  Recordings.
+  """
+
+  use Ecto.Schema
+  import Ecto.Changeset
+  import SweetXml
+
+  @primary_key false
+  @derive Jason.Encoder
+  @required [:job_token]
+  @optional []
+
+  embedded_schema do
+    field(:job_token, :string)
+
+    embeds_one :job_configuration, JobConfiguration, primary_key: false, on_replace: :update do
+      @derive Jason.Encoder
+      field(:recording_token, :string)
+      field(:mode, :string)
+      field(:priority, :string)
+
+      embeds_one :source, Source, primary_key: false, on_replace: :update do
+        @derive Jason.Encoder
+        field(:auto_create_receiver, :boolean)
+
+        embeds_one :source_token, SourceToken, primary_key: false, on_replace: :update do
+          @derive Jason.Encoder
+          field(:token, :string)
+        end
+
+        embeds_many :tracks, Tracks, primary_key: false, on_replace: :delete do
+          @derive Jason.Encoder
+          field(:source_tag, :string)
+          field(:destination, :string)
+        end
+      end
+    end
+  end
+
+  def parse(nil), do: nil
+  def parse([]), do: nil
+
+  def parse(doc) do
+    xmap(
+      doc,
+      job_token: ~x"./tt:JobToken/text()"so,
+      job_configuration: ~x"./tt:JobConfiguration"eo |> transform_by(&parse_job_configuration/1)
```

Comment:
should `tt:JobConfiguration` be `trc:JobConfiguration`?

Reply:
nope, as you can see here https://github.com/hammeraj/onvif/blob/po/exportRecordedData/test/recording/fixture/get_recording_jobs_success.xml#L58 it is `tt`

Reply:
Interesting, I looked through the docs and it appears that the JobConfiguration in the Create response _should_ also be in the tt namespace but there are inconsistencies with the Onvif documentation